src/sentry/api/endpoints/organization_code_mapping_codeowners.py

```diff
@@ -25,18 +34,11 @@ def convert_args(self, request: Request, organization_slug, config_id, *args, **
 
         return (args, kwargs)
 
-    def _get_codeowner_contents(self, config):
-        integration = config.organization_integration.integration
-        install = integration.get_installation(config.organization_integration.organization_id)
-        return install.get_codeowner_file(config.repository, ref=config.default_branch)
-
     def get(self, request: Request, config_id, organization, config) -> Response:
-        if not config.organization_integration:
-            return self.respond(
-                {"error": "No associated integration."}, status=status.HTTP_400_BAD_REQUEST
-            )
-
-        codeowner_contents = self._get_codeowner_contents(config)
+        try:
+            codeowner_contents = get_codeowner_contents(config)
+        except Exception as e:
```

Comment:
@NisanthanNanthakumar It would be better to try to catch the expected errors you're raising rather than catching any possible error which might actually mask a real problem. And in some cases, such as a missing `organization_integration` row, a `404` response would be more appropriate.